packages/formation/js/sidenav.js

```diff
@@ -0,0 +1,157 @@
+        this.closeControl.focus();
+      }
+    }
+  }
+  getAccordionButtons(node) {
+    return node.querySelectorAll('.usa-accordion-button');
+  }
+  loadKeyNodes(element) {
+    this.menu = document.querySelector(
+      `#${element.getAttribute('aria-controls')}`,
+    );
+    this.tabbableElements = getTabbableElements(this.menu);
+    this.accordionButtons = this.getAccordionButtons(this.menu);
+    this.lastAccordionButton = this.accordionButtons[
+      this.accordionButtons.length - 1
+    ];
+    this.closeControl = this.tabbableElements[0];
+    this.lastTabbableElement = this.tabbableElements[
+      this.tabbableElements.length - 1
+    ];
+  }
+  openMenu(trigger) {
+    this.loadKeyNodes(trigger);
+    this.menu.classList.add('va-sidebarnav--opened');
+    this.menu.setAttribute('aria-hidden', 'false');
+    document
+      .getElementsByClassName('va-btn-sidebarnav-trigger')[0]
+      .setAttribute('hidden', 'true');
+    document.getElementsByTagName('body')[0].style.overflow = 'hidden';
+    this.closeMenu(trigger);
+    if (trigger.className.includes('va-btn-sidebarnav-trigger')) {
+      // main trigger, switch focus to close button
+      this.closeControl.focus();
+    }
+    // capture the focus
+    if (this.lastAccordionButton) {
+      this.lastAccordionButton.addEventListener(
+        'keydown',
+        this.checkAccordionFocus,
+      );
+    }
+    this.closeControl.addEventListener('keydown', this.captureFocus);
+    this.lastTabbableElement.addEventListener('keydown', this.captureFocus);
+  }
+  removeListeners() {
+    if (this.lastAccordionButton) {
+      this.lastAccordionButton.removeEventListener(
+        'keydown',
+        this.checkAccordionFocus,
+      );
+    }
+    this.closeControl.removeEventListener('keydown', this.captureFocus);
+    this.lastTabbableElement.removeEventListener('keydown', this.captureFocus);
+  }
+
+  closeMenu() {
+    const close = this.menu.querySelector('.va-sidebarnav-close');
+    close.addEventListener('click', () => {
+      this.menu.classList.remove('va-sidebarnav--opened');
+      this.menu.setAttribute('aria-hidden', 'true');
+      document.getElementsByTagName('body')[0].style.overflow = 'initial';
+      document
+        .getElementsByClassName('va-btn-sidebarnav-trigger')[0]
+        .removeAttribute('hidden');
+      this.removeListeners();
+    });
+  }
+}
+
+export function addActiveState() {
+  // Add active state for Drupal-based side nav
+  const sideNav = document.querySelector(
+    '#va-detailpage-sidebar[data-drupal-sidebar]',
+  );
+  if (sideNav) {
+    const current = document.getElementsByClassName('usa-current')[0];
+    const parent = current.closest('.usa-accordion-content')
+      .previousElementSibling;
+    const ariaExpanded =
+      parent.getAttribute('aria-expanded') === 'false' ? 'true' : 'false';
+    parent.setAttribute('aria-expanded', ariaExpanded);
+  }
+}
+
+document.addEventListener('DOMContentLoaded', () => {
```

Comment:
I think we want to remove these addEventListener calls and combine them into one exported function:

```js
export function addSidenavListeners() {
  // eslint-disable-next-line no-new
  new SideBarMenu(document.querySelectorAll('.va-btn-sidebarnav-trigger'));
  addActiveState();
}
```